src/main/java/org/gitlab4j/api/DiscussionsApi.java

```diff
@@ -657,6 +658,40 @@ public Discussion createCommitDiscussion(
         return (response.readEntity(Discussion.class));
     }
 
+    /**
+     * Creates a new discussion to a single project commit. This is similar to creating
+     * a note but other comments (replies) can be added to it later.
+     *
+     * <pre><code>GitLab Endpoint: POST /projects/:id/repository/commits/:commit_sha/discussions</code></pre>
+     *
+     * @param projectIdOrPath projectIdOrPath the project in the form of an Long(ID), String(path), or Project instance
+     * @param commitSha the commit SHA to create the discussion for
+     * @param body the content of a discussion
+     * @param createdAt date the discussion was created (requires admin or project/group owner rights) (Optional)
+     * @return a Discussion instance containing the newly created discussion
+     * @throws GitLabApiException if any exception occurs during execution
+     */
+    public Discussion createCommitDiscussion(Object projectIdOrPath, String commitSha, String body, Date createdAt)
+            throws GitLabApiException {
+
+        GitLabApiForm formData =
+                new GitLabApiForm().withParam("body", body, true);
+        if (createdAt != null) {
+            formData.withParam("created_at", ISO8601.toString(createdAt));
```

Comment:
Null check technically not necessary. Value could be annotated @Nullable but no current libraries support it and did not want to add one myself.  

Would recommend:
`import javax.annotation.Nonnull;`
from:
`\com\google\code\findbugs\jsr305`